tests/dates/test_BaseDateTransformer.py

```diff
@@ -0,0 +1,153 @@
+                datetime.date(2015, 11, 10),
+                datetime.date(2015, 7, 23),
+            ],
+            "datetime_col_1": [
+                datetime.datetime(1993, 9, 27, tzinfo=datetime.timezone.utc),
+                datetime.datetime(2000, 3, 19, tzinfo=datetime.timezone.utc),
+                datetime.datetime(2018, 11, 10, tzinfo=datetime.timezone.utc),
+                datetime.datetime(2018, 10, 10, tzinfo=datetime.timezone.utc),
+                datetime.datetime(2018, 10, 10, tzinfo=datetime.timezone.utc),
+                datetime.datetime(2018, 10, 10, tzinfo=datetime.timezone.utc),
+                datetime.datetime(2018, 12, 10, tzinfo=datetime.timezone.utc),
+                datetime.datetime(
+                    1985,
+                    7,
+                    23,
+                    tzinfo=datetime.timezone.utc,
+                ),
+            ],
+            "datetime_col_2": [
+                datetime.datetime(2020, 5, 1, tzinfo=datetime.timezone.utc),
+                datetime.datetime(2019, 12, 25, tzinfo=datetime.timezone.utc),
+                datetime.datetime(2018, 11, 10, tzinfo=datetime.timezone.utc),
+                datetime.datetime(2018, 11, 10, tzinfo=datetime.timezone.utc),
+                datetime.datetime(2018, 9, 10, tzinfo=datetime.timezone.utc),
+                datetime.datetime(2015, 11, 10, tzinfo=datetime.timezone.utc),
+                datetime.datetime(2015, 11, 10, tzinfo=datetime.timezone.utc),
+                datetime.datetime(2015, 7, 23, tzinfo=datetime.timezone.utc),
+            ],
+        },
+    )
+
+
+class DatesMixinTransformTests:
+    """Generic tests for Dates Transformers"""
+
+    @pytest.mark.parametrize(
+        ("columns, datetime_col, date_col"),
+        [
+            (["date_col_1", "datetime_col_2"], 1, 0),
+            (["datetime_col_1", "date_col_2"], 0, 1),
+        ],
+    )
+    def test_mismatched_datetypes_error(
+        self,
+        columns,
+        datetime_col,
+        date_col,
+        uninitialized_transformers,
+    ):
+        "Test that transform raises an error if one column is a date and one is datetime"
+
+        x = uninitialized_transformers[self.transformer_name](
+            columns=columns,
+            new_column_name="c",
+        )
+
+        df = create_date_diff_different_dtypes()
+        # types don't seem to come out of the above function as expected, hard enforce
+        for col in ["date_col_1", "date_col_2"]:
+            df[col] = pd.to_datetime(df[col]).dt.date
+
+        for col in ["datetime_col_1", "datetime_col_2"]:
+            df[col] = pd.to_datetime(df[col])
+
+        present_types = (
+            {"datetime64", "date"} if datetime_col == 0 else {"date", "datetime64"}
+        )
+        msg = rf"Columns fed to datetime transformers should be \['datetime64', 'date'\] and have consistent types, but found {present_types}. Please use ToDatetimeTransformer to standardise"
```

Comment:
N: Is this message a bit misleading? Just to check my understanding we want both columns to have a consistent type, but the 'should be \['datetime64', 'date'\]' suggests that we want one dt64 and one dt column?

Reply:
This is set up to read the allowed types argument passed to the method so this error message might say datetime64 or date in it and this expression will accept either.

Reply:
Ahh okay thanks :)